### PR TITLE
fix: conditional type exports (T1177805)

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**
Testing with https://github.com/arethetypeswrong/arethetypeswrong.github.io, before/after:

image

image

### CODE REVIEW

**Review by const314 — APPROVED** [review]

File: scripts/utils.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { existsSync, copyFileSync } from 'fs';
+
+const getCommonJsTypesPath = dtsPath => dtsPath.replace(/.ts$/, '.cts');
```
Comment (timbset) [inline scripts/utils.js:3]:
```suggestion
const getCommonJsTypesPath = dtsPath => dtsPath.replace(/\.ts$/, '.cts');
```

In your version a dot means any symbol

**Review by timbset — CHANGES_REQUESTED** [review]

**Review by timbset — APPROVED** [review]

File: scripts/merge-dts.js [hunk 2]
```diff
@@ -21,8 +21,8 @@ export default (packageDirectory, skipBundle = false) => {
   const dtsPath = join(packageDirectory, 'dist', 'dts');
 
   if (!skipBundle) {
-    const pkg = JSON.parse(readFileSync(join(packageDirectory, 'package.json')));
-    const dtsOutFile = join(packageDirectory, pkg.types);
+    const pkg = getPackageInfo(packageDirectory);
+    const dtsOutFile = getDtsOutFile(packageDirectory, pkg.types);
```
Comment (timbset) [inline scripts/merge-dts.js:25]: Looks like an overcomplicating. What's the difference between your function and using `join`?

File: scripts/utils.js [hunk 3]
```diff
@@ -7,3 +8,7 @@ export const copyCommonJsTypes = (dtsPath) => {
     copyFileSync(dtsPath, getCommonJsTypesPath(dtsPath));
   }
 };
+
+export const getPackageInfo = packageDir => JSON.parse(readFileSync(join(packageDir, 'package.json')));
+
+export const getDtsOutFile = (packageDir, typesDir) => join(packageDir, typesDir);
```
Comment (timbset) [inline scripts/utils.js:14]: We've simply duplicated using of `join`. Remove this function

File: scripts/merge-dts.js [hunk 4]
```diff
@@ -21,8 +21,8 @@ export default (packageDirectory, skipBundle = false) => {
   const dtsPath = join(packageDirectory, 'dist', 'dts');
 
   if (!skipBundle) {
-    const pkg = JSON.parse(readFileSync(join(packageDirectory, 'package.json')));
-    const dtsOutFile = join(packageDirectory, pkg.types);
+    const pkg = getPackageInfo(packageDirectory);
```
Comment (timbset) [inline scripts/merge-dts.js:24]: I think it is better to split this on steps instead of extracting this logic into util function

**Review by timbset — CHANGES_REQUESTED** [review]

**Pull request merged**